types/wlr_surface.c

```diff
@@ -303,8 +313,12 @@ static void surface_state_move(struct wlr_surface_state *state,
 	surface_state_copy(state, next);
 
 	if (next->committed & WLR_SURFACE_STATE_BUFFER) {
+		if (next->buffer) {
+			state->buffer = wlr_buffer_lock(next->buffer);
+		}
 		surface_state_set_buffer(state, next->buffer_resource);
 		surface_state_reset_buffer(next);
+		next->buffer = NULL;
```

Comment:
Here we unset `next->buffer` without decrementing its ref'count.

Reply:
Feels a bit weird to lock it when copying it to then unlock it a few lines beneath, but I get why.

Reply:
If you lock it then unlock if immediately, it's possible to simplify to just "transfer" the ref'count.

e.g. it's perfectly fine to change

```c
b = wlr_buffer_lock(a);
wlr_buffer_unlock(a);
a = NULL;
```

to

```c
b = a;
a = NULL;
```